File: easy/problems/0389-find-the-difference/solution.ts
function findTheDifference(s: string, t: string): string {
    let result = 0;

    for (const char of s) {
        result ^= char.charCodeAt(0);
    }

    for (const char of t) {
        result ^= char.charCodeAt(0);
    }

    return String.fromCharCode(result);
}

File: easy/problems/0389-find-the-difference/solution.py
def findTheDifference(s, t):
    """
    Find the extra character in string t that's not in s.

    Args:
        s: str - original string
        t: str - string with one extra character

    Returns:
        str - the extra character
    """
    result = 0

    for char in s:
        result ^= ord(char)

    for char in t:
        result ^= ord(char)

    return chr(result)
